src/sql/workbench/services/objectExplorer/browser/connectionTreeAction.ts

```diff
@@ -290,7 +290,8 @@ export class DeleteConnectionAction extends Action {
 		const deleteConnectionConfirmationNo = localize('deleteConnectionConfirmationNo', "No");
 
 		if (this.element instanceof ConnectionProfile) {
-			const modalResult = await this._dialogService.show(Severity.Warning, localize('deleteConnectionConfirmation', "Are you sure you want to delete connection '{0}'?", this.element.connectionName),
+			const name = this.element.connectionName ? this.element.connectionName : this.element.serverName;
```

Comment:
nit:
```suggestion
			const name = this.element.connectionName || this.element.serverName;
```

Reply:
That works too? That's a really clever way of doing the assignment.

Reply:
Yes, but you have to be careful since `||` will use the RHS for any falsy value of the LHS (such as an empty string or 0). `??` is for when you only want it for undefined/null values. 

https://www.typescriptlang.org/play?#code/PTAEHUFMBsGMHsC2lQBd5oBYoCoE8AHSAZVgCcBLA1UABWgEM8BzM+AVwDsATAGiwoBnUENANQAd0gAjQRVSQAUCEmYKsTKGYUAbpGF4OY0BoadYKdJMoL+gzAzIoz3UNEiPOofEVKVqAHSKymAAmkYI7NCuqGqcANag8ABmIjQUXrFOKBJMggBcISGgoAC0oACCbvCwDKgU8JkY7p7ehCTkVDQS2E6gnPCxGcwmZqDSTgzxxWWVoASMFmgYkAAeRJTInN3ymj4d-jSCeNsMq-wuoPaOltigAKoASgAywhK7SbGQZIIz5VWCFzSeCrZagNYbChbHaxUDcCjJZLfSDbExIAgUdxkUBIursJzCFJtXydajBBCcQQ0BRUgCMoAAvKAAOQACRg0HgzNAAB8eSzwPAyNFmQBuRQUqlofSoABMjJZ3L5AqFIvFkupMoAzAquNxIMkMpBXMrmYLhdwxRLGoJ4O4ApzmAAKGmoWkASmtlLtkAd8GdrtlnslPr9Ae1nuCKhwzQ8ZC8iCFzmB7Gpd0YnGY7AYzEg-Fg0HUiQYwL0Ii8ACIAKKrBiIBb6CtJbEV8AOVDM4QAOUgEgrQRUAHkvmR3oI81pIEdUI4FK5pHhQE5Ezphlh9CgEIhocJLl9JKrXKJDPj5owWGw9UESkA